### PR TITLE
Fix OGR database connection dialog

### PR DESCRIPTION
**Pull request by agiudiceandrea**
## Description

Fixes the following unreported issue about the Data Source Manager -> Vector -> Database -> "New OGR Database Connection" dialog window:

- the database type combo box incorrectly contains an empty item

image

- the dialog always opens with the first database type item selected, even if a different database type is selected in the Data Source Manager -> Vector -> Database dialog window before clicking on the "New" button

image

### CODE REVIEW

**Review by lbartoletti — APPROVED** [review]

File: src/gui/ogr/qgsnewogrconnection.cpp [hunk 1]
```diff
@@ -54,11 +54,18 @@ QgsNewOgrConnection::QgsNewOgrConnection( QWidget *parent, const QString &connTy
   const QStringList dbDrivers = QgsProviderRegistry::instance()->databaseDrivers().split( ';' );
   for ( int i = 0; i < dbDrivers.count(); i++ )
   {
-    const QString dbDrive = dbDrivers.at( i );
-    cmbDatabaseTypes->addItem( dbDrive.split( ',' ).at( 0 ) );
+    const QString dbDriver = dbDrivers.at( i );
+    if ( ( !dbDriver.isEmpty() ) && ( !dbDriver.isNull() ) )
```
Comment (nyalldawson) [inline src/gui/ogr/qgsnewogrconnection.cpp:58]:
```suggestion
    if ( !dbDriver.isEmpty() )
```

It's not possible for this string to be null

Comment (agiudiceandrea) [reply src/gui/ogr/qgsnewogrconnection.cpp]: https://github.com/qgis/QGIS/blob/8491e83166ea173fe74ef046dfed0f0f7dbc2e0c/src/gui/providers/ogr/qgsogrsourceselect.cpp#L81-L82
https://github.com/qgis/QGIS/blob/8491e83166ea173fe74ef046dfed0f0f7dbc2e0c/src/gui/providers/ogr/qgsogrsourceselect.cpp#L90-L91
https://github.com/qgis/QGIS/blob/8491e83166ea173fe74ef046dfed0f0f7dbc2e0c/src/gui/providers/ogr/qgsogrsourceselect.cpp#L101-L102
should also be fixed?

Comment (nyalldawson) [reply src/gui/ogr/qgsnewogrconnection.cpp]: Yes, there's no way QString::split will return a null string. It'll always be empty.

Comment (agiudiceandrea) [reply src/gui/ogr/qgsnewogrconnection.cpp]: Done.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**